### ISSUE TITLE
Android notification channel

### ISSUE THREAD

**Issue by Freddyvdh** [thudugala/Plugin.LocalNotification#24]
Hi,

thank you for this great plugin and accepting the PR.

I'm still testing with the channel. When I change the description of the channel name, the settings are set back to default. The NotificationManager can retrieve the channel if it exists with `GetNotificationChannel(string channelId)`.

I checked the android [docs](https://developer.android.com/training/notify-user/channels):
```kotlin
CharSequence name = getString(R.string.channel_name);
String description = getString(R.string.channel_description);
int importance = NotificationManager.IMPORTANCE_DEFAULT;
NotificationChannel channel = new NotificationChannel(CHANNEL_ID, name, importance);
channel.setDescription(description);
// Register the channel with the system; you can't change the importance
// or other notification behaviors after this
NotificationManager notificationManager = getSystemService(NotificationManager.class);
notificationManager.createNotificationChannel(channel);
```

Priority can't bet set for each NotificationRequest, it is only working for the first. I don't know what is better: Extend INotificationService with and AndroidCreateChannel Method and remove Priority and ChannelDescription from AndroidOptions, or creating the channel with the first NotificationRequest. With second option, the comment should be extended.

I'm wondering why the channel is recreate, when I change the channel description.

**Comment by thudugala** [comment]
@Freddyvdh I like the second option "Create the channel with the first NotificationRequest"

**Comment by pikausp** [comment]
Personally, I believe it doesn't make sense to expose properties for each notification request that will be ignored in subsequent request.

**Comment by thudugala** [comment]
@pikausp and @Freddyvdh Can not add AndroidCreateChannel to INotificationService as it is not for iOS.
Maybe in NotificationCenter in Platform.Droid.

**Comment by thudugala** [comment]
@pikausp and @Freddyvdh now Notification Channel is created once when app start.

```cs
public class MainActivity : global::Xamarin.Forms.Platform.Android.FormsAppCompatActivity
{
    protected override void OnCreate(Bundle savedInstanceState)
    {
        ....
        LoadApplication(new App());
        NotificationCenter.Init(this.Intent);
    }
}
```

**Comment by Freddyvdh** [comment]
@thudugala it looks good, but it doesn't work with two or more channels, which I wan't to use.

I would add the channel name again to AndroidOptions, if not set it is general.

Remove `NotificationChannelId` as static from NotificationCenter, and make it a local variable in `CreateNotificationChannel`.

And I would change the creation to:

`var channel = notificationManager.GetNotificationChannel(NotificationChannelId);
            if (channel == null)
            {
                // you can't change the importance or other notification behaviors after this.
                // once you create the channel, you cannot change these settings and
                // the user has final control of whether these behaviors are active.
                channel = new NotificationChannel(NotificationChannelId, request.Name, request.Importance)
                {
                    Description = request.Description,
                    Group = request.Group,
                    LightColor = request.LightColor,
                    LockscreenVisibility = request.LockscreenVisibility,
                };
            }`

**Comment by thudugala** [comment]
@Freddyvdh Creating an existing notification channel with its original values performs no operation, so it's safe to call this code when starting an app.

https://developer.android.com/training/notify-user/channels#CreateChannel

**Comment by thudugala** [comment]
@Freddyvdh now Multiple Notification Channels can be created once when app start.

```cs
public class MainActivity : global::Xamarin.Forms.Platform.Android.FormsAppCompatActivity
{
    protected override void OnCreate(Bundle savedInstanceState)
    {
        ....
        // Must create a Notification Channel when API >= 26
        // Can created multiple Notification Channels with different names.
        NotificationCenter.CreateNotificationChannel();
        ....
        LoadApplication(new App());    
        ....    
        NotificationCenter.NotifyNotificationTapped(Intent);
    }
}
```
try LocalNotification.Sample.sln in Sample/Direct folder

**Comment by Freddyvdh** [comment]
Works great.
I had an error on compiling the android sample and had to enable [multi dex](https://github.com/xamarin/Xamarin.Forms/issues/4825).

**Comment by Freddyvdh** [comment]
One, I hope last thing ;)

For localization it would be great, if the channel id, name and description could be set in the NotificationChannelRequest. The channel id must be given in the AndroidOptions.

**Comment by thudugala** [comment]
@Freddyvdh try LocalNotification.Sample.sln in Sample/Direct folder. 

Now channel id can be set in NotificationChannelRequest.Id, if set, the channel id must also be set in NotificationRequest.AndroidOptions.ChannelId

**Comment by thudugala** [comment]
Released [4.0.0](https://www.nuget.org/packages/Plugin.LocalNotification/4.0.0)
